### PR TITLE
Fontkit.js support

### PR DESCRIPTION
**Pull request by jchavarri**
~~Blocked on https://github.com/bryphe/reason-fontkit/pull/6.~~ 

TODO:

- [x] Remove the `resolution` to `../reason-fontkit` and point it to the latest version after the PR above is merged.

### CODE REVIEW

**Comment by bryphe** [comment]
Thanks @jchavarri ! Once we have bryphe/reason-fontkit#6 in, I'll publish a new NPM version 👍

**Comment by bryphe** [comment]
Just tried this out in combination with the fix in `reason-fontkit` - worked great! 👍

**Comment by jchavarri** [comment]
@bryphe Just merged https://github.com/revery-ui/reason-fontkit/pull/6 🎉 Please let me know when a new version is ready and I can pick this PR up to update the dependencies 😄  Thanks!

File: src/UI/Revery_UI.re [hunk 1]
```diff
@@ -123,6 +123,10 @@ let start =
   let container = UiReact.createContainer(rootNode);
   let ui: uiContainer = {window, rootNode, container, options: createOptions};
 
+  let _ = Reactify.Event.subscribe(FontCache.onFontLoaded, () => {
```
Comment (bryphe) [inline src/UI/Revery_UI.re]: Cool, thanks for implementing this strategy 👍

**Comment by bryphe** [comment]
@jchavarri - sorry about the build issue! Thanks for the heads up - I checked the Webhooks and saw a few that failed (must be a transient issue on the Azure CI side) - I 'redelivered' those and looks like it picked it up now. 🤞

**Comment by OhadRau** [comment]
FYI reason-fontkit is failing for me on 2.0.1 (in WSL):

```
info building reason-fontkit@1.2.0: starting                                                                            
error: build failed                                                                                                       
build log:                                                                                                                
# esy-build-package: building: reason-fontkit@1.2.0                                                                     
# esy-build-package: running: 'dune' 'build' '--root' '.'                                                                   
ocamlopt src/fontkit.cmxs (exit 2)                                                                                  
(cd _build/default && /home/ohadrau/.esy/3__________________________________________________________________/i/ocaml-4.6.9-ff9c6a38/bin/ocamlopt.opt -w @a-4-29-40-41-42-44-45-48-58-59-60-40 -strict-sequence -strict-formats -short-paths -keep-locs -g -shared -linkall -I src -o src/fontkit.cmxs src/fontkit.cmxa)                                                 
/usr/bin/ld: /home/ohadrau/.esy/3__________________________________________________________________/i/esy_harfbuzz-1.9.1003-6cbce9bc/lib/libharfbuzz.a(libharfbuzz_la-hb-ot-layout.o): relocation R_X86_64_PC32 against symbol `hb_ot_layout_language_find_feature' can not be used when making a shared object; recompile with -fPIC                                  
 
/usr/bin/ld: final link failed: Bad value                                                                               
collect2: error: ld returned 1 exit status                                                                              
File "caml_startup", line 1:                                                                                            
Error: Error during linking                                                                                             
error: command failed: 'dune' 'build' '--root' '.'                                                                      
esy-build-package: exiting with errors above...                                                                                                                                                                                               
building reason-fontkit@1.2.0  
```

**Comment by jchavarri** [comment]
hey @bryphe I merged the latest master but the same error is still there. I am not sure if it's related to the latest changes in `fontkit` that switched to prebuilt dependency for esy-freetype2:

[Build 49](https://dev.azure.com/revery-ui/revery/_build/results?buildId=49), with reason-fontkit 2.0.0:
```
2018-12-26T18:53:39.9337562Z error: build failed
2018-12-26T18:53:39.9338553Z   build log:
2018-12-26T18:53:39.9339426Z     # esy-build-package: building: reason-fontkit@2.0.0
2018-12-26T18:53:39.9339928Z     # esy-build-package: running: 'dune' 'build' '--root' '.'
2018-12-26T18:53:39.9340155Z              gcc src/fontkit.o
2018-12-26T18:53:39.9340616Z     fontkit.cpp: In function 'hb_font_t* get_font_ot(const char*, int)':
2018-12-26T18:53:39.9341116Z     fontkit.cpp:74:35: warning: ignoring return value of 'size_t fread(void*, size_t, size_t, FILE*)', declared with attribute warn_unused_result [-Wunused-result]
2018-12-26T18:53:39.9341347Z            fread(data, length, 1, file);
2018-12-26T18:53:39.9341510Z                                        ^
2018-12-26T18:53:39.9341647Z         ocamlopt src/fontkit.cmxs (exit 2)
2018-12-26T18:53:39.9342816Z     (cd _build/default && /home/vsts/.esy/3_____________________________________________________________________/i/ocaml-4.6.9-ff9c6a38/bin/ocamlopt.opt -w @a-4-29-40-41-42-44-45-48-58-59-60-40 -strict-sequence -strict-formats -short-paths -keep-locs -g -shared -linkall -I src -o src/fontkit.cmxs src/fontkit.cmxa)
2018-12-26T18:53:39.9343659Z     /usr/bin/ld: /home/vsts/.esy/3_____________________________________________________________________/i/esy_harfbuzz-1.9.1003-6cbce9bc/lib/libharfbuzz.a(libharfbuzz_la-hb-blob.o): relocation R_X86_64_32 against `_hb_Null_hb_blob_t' can not be used when making a shared object; recompile with -fPIC
2018-12-26T18:53:39.9344539Z     /home/vsts/.esy/3_____________________________________________________________________/i/esy_harfbuzz-1.9.1003-6cbce9bc/lib/libharfbuzz.a: error adding symbols: Bad value
2018-12-26T18:53:39.9344805Z     collect2: error: ld returned 1 exit status
2018-12-26T18:53:39.9345024Z     File "caml_startup", line 1:
2018-12-26T18:53:39.9345180Z     Error: Error during linking
2018-12-26T18:53:39.9345580Z     error: command failed: 'dune' 'build' '--root' '.'
2018-12-26T18:53:39.9345993Z     esy-build-package: exiting with errors above...
2018-12-26T18:53:39.9346182Z     
2018-12-26T18:53:39.9346548Z   building reason-fontkit@2.0.0
```

[Build 59](https://dev.azure.com/revery-ui/revery/_build/results?buildId=59), with reason-fontkit 2.0.1:
```
2018-12-27T09:06:39.5675263Z error: build failed
2018-12-27T09:06:39.5676588Z   build log:
2018-12-27T09:06:39.5677493Z     # esy-build-package: building: reason-fontkit@2.0.1
2018-12-27T09:06:39.5678096Z     # esy-build-package: running: 'dune' 'build' '--root' '.'
2018-12-27T09:06:39.5678147Z              gcc src/fontkit.o
2018-12-27T09:06:39.5678792Z     fontkit.cpp: In function 'hb_font_t* get_font_ot(const char*, int)':
2018-12-27T09:06:39.5679129Z     fontkit.cpp:74:35: warning: ignoring return value of 'size_t fread(void*, size_t, size_t, FILE*)', declared with attribute warn_unused_result [-Wunused-result]
2018-12-27T09:06:39.5679191Z            fread(data, length, 1, file);
2018-12-27T09:06:39.5679269Z                                        ^
2018-12-27T09:06:39.5679308Z         ocamlopt src/fontkit.cmxs (exit 2)
2018-12-27T09:06:39.5679731Z     (cd _build/default && /home/vsts/.esy/3_____________________________________________________________________/i/ocaml-4.6.9-ff9c6a38/bin/ocamlopt.opt -w @a-4-29-40-41-42-44-45-48-58-59-60-40 -strict-sequence -strict-formats -short-paths -keep-locs -g -shared -linkall -I src -o src/fontkit.cmxs src/fontkit.cmxa)
2018-12-27T09:06:39.5680189Z     /usr/bin/ld: /home/vsts/.esy/3_____________________________________________________________________/i/esy_harfbuzz-1.9.1003-6cbce9bc/lib/libharfbuzz.a(libharfbuzz_la-hb-blob.o): relocation R_X86_64_32 against `_hb_Null_hb_blob_t' can not be used when making a shared object; recompile with -fPIC
2018-12-27T09:06:39.5680525Z     /home/vsts/.esy/3_____________________________________________________________________/i/esy_harfbuzz-1.9.1003-6cbce9bc/lib/libharfbuzz.a: error adding symbols: Bad value
2018-12-27T09:06:39.5680612Z     collect2: error: ld returned 1 exit status
2018-12-27T09:06:39.5680652Z     File "caml_startup", line 1:
2018-12-27T09:06:39.5680710Z     Error: Error during linking
2018-12-27T09:06:39.5681925Z     error: command failed: 'dune' 'build' '--root' '.'
2018-12-27T09:06:39.5682184Z     esy-build-package: exiting with errors above...
2018-12-27T09:06:39.5682446Z     
2018-12-27T09:06:39.5682747Z   building reason-fontkit@2.0.1
```

Although, it could also be a caching issue?

**Comment by bryphe** [comment]
Checked and builds are passing with the latest version - will bring this in now so we can get everything up-to-date 👍 Thanks for all your work here @jchavarri !

**Comment by jchavarri** [comment]
@bryphe Thank you for fixing all the CI side! 🚀

**Pull request merged**